test/targets.cpp

```diff
@@ -50,4 +51,41 @@ TEST_CASE(targets)
     EXPECT(ts.size() >= 1);
 }
 
+TEST_CASE(concurrent_targets)
+{
+    std::vector<std::thread> threads;
+#ifdef HAVE_GPU
+    std::string target_name = "gpu";
+#elif defined(HAVE_CPU)
+    std::string target_name = "cpu";
+#elif defined(HAVE_FPGA)
+    std::string target_name = "fpga";
+#else
+    std::string target_name = "ref";
+#endif
+
+    auto n_threads = std::thread::hardware_concurrency() * 4;
+
+    for(auto i = 0u; i < n_threads; i++)
+    {
+        auto thread_body = [&target_name]() {
+
+            // remove all existing targets, if any
+            std::vector<std::string> target_list = migraphx::get_targets();
+            for(auto tt : target_list)
```

Comment:
This new loop breaks the test now.  Multiple threads registering and unregistering a target with the same name is not safe.  Apparently it's not the target_map that's a problem but the underlying handles.  Trying to figure out how to protect against this, or if it's necessary.

Reply:
Followup:  I disabled this extra test but when I tested the existing fixes with amdinfer, the Inference Server, the segmentation violation still occurs.  I'm going to recommend we commit this PR as is but continue to search for the root cause of Issue #2208.